lib/ipip.c

```diff
@@ -113,7 +113,7 @@ encapsulate(struct rte_mbuf *pkt, uint8_t priority,
 			sizeof(outer_ip6hdr->dst_addr));
 
 		outer_ip6hdr->payload_len = rte_cpu_to_be_16(pkt->data_len
-			- sizeof(struct ipv6_hdr));
```

Comment:
1. Given that we are fixing the payload length, shouldn't we replace `pkt->data_len` with `pkt->pkt_len`? If so, we should review the IPv4 case as well.

2. Shouldn't we adopt the macros `rte_pktmbuf_pkt_len()` and `rte_pktmbuf_data_len()`?

Reply:
1. Recall that a single frame can be represented by multiple `struct rte_mbuf`s in a linked list if the frame is too large to fit into one `struct rte_mbuf` segment. Because of this, `data_len` is the length of the single given `struct rte_mbuf` segment, and `pkt_len` is the length of all of the segments in the chain that the `struct rte_mbuf` is a part of.

The only difference between `data_len` and `pkt_len` happens when the `struct rte_mbuf` is not large enough to fit the frame. We set our mbufs to have room for 2048 bytes of data, and to this point have assumed that we won't have frames bigger than 2048 bytes. Removing this assumption requires handling #93.

Whenever we need the length of a packet, we have used `data_len` throughout the code. The only time we touch `pkt_len` is when we create new packets (and set `pkt_len` = `data_len`).

2. I'm OK with using these macros, but if we do then we should probably replace all 50-60 occurrences of reads of these variables with the macros.

Reply:
I've added item 2 to #93, so let's postpone item 1 and 2 to when we address that issue.